### PR TITLE
fix: structured inputs for CudaGraphsTorchTensorRTModule

### LINKED ISSUE #3406: 🐛 [Bug] Encountered bug when using Torch-TensorRT
##  Bug Description

When using CudaGraph, if there is a graph break resulting a pytorch subgraph, and if the input has a nested dictionary, cuda graph breaks.
In the following example, the number of input is 2 in the forward function, but 3 in the graph module, which gives an error.

## To Reproduce

```
import torch
import torch_tensorrt

class TestModel(torch.nn.Module):
    def forward(self, x, additional_param: dict):
        x = x + additional_param['y']
        return x  * additional_param['z'] + 5

device = "cuda:0"
inputs=(torch.rand(1).to(device),)
kwarg_inputs={
    "additional_param": {
        'y':torch.rand(1).to(device), 
        'z':torch.rand(1).to(device), 
    }
}
model = TestModel().to(device)
compiled_model = torch_tensorrt.compile(
    model,
    ir="dynamo",
    arg_inputs=inputs,
    kwarg_inputs=kwarg_inputs,
    min_block_size=1,
    torch_executed_ops={"torch.ops.aten.mul.Tensor"}
)

with torch_tensorrt.runtime.enable_cudagraphs(
    compiled_model
) as cudagraphs_module:
    cudagraphs_module(*inputs, **kwarg_inputs)
```

## Environment

> Build information about Torch-TensorRT can be found by turning on debug messages

 - Torch-TensorRT Version (e.g. 1.0.0): nightly
 - PyTorch Version (e.g. 1.0): nightly
 - CPU Architecture: x86
 - OS (e.g., Linux): Linux
 - How you installed PyTorch (`conda`, `pip`, `libtorch`, source): pip
 - Build command you used (if compiling from source):
 - GPU models and configuration: A40

### PR DESCRIPTION
**Pull request by keehyuna**
# Description

Add support for structured inputs in CudaGraphsTorchTensorRTModule using torch's tree utilities to properly handle both args and kwargs input

Fixes #3406 

## Type of change

Please delete options that are not relevant and/or add your own.

- Bug fix (non-breaking change which fixes an issue)

# Checklist:

- [ ] My code follows the style guidelines of this project (You can use the linters)
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas and hacks
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests to verify my fix or my feature
- [ ] New and existing unit tests pass locally with my changes
- [ ] I have added the relevant labels to my PR in so that relevant reviewers are notified

### CODE REVIEW

File: tests/py/dynamo/runtime/test_004_weight_streaming.py [hunk 1]
```diff
@@ -302,45 +303,62 @@ def __init__(self):
...
+                    "value2": torch.tensor(1.2).to("cuda"),
+                },
+                "e": [torch.rand(1).to("cuda"), torch.rand(1).to("cuda")],
+            }
+            input_list.append((args, kwargs))
+
+        kwarg_torchtrt_input = prepare_inputs(input_list[0][1])
+
+        compile_spec = {
+            "inputs": [
```
Comment (narendasan) [inline tests/py/dynamo/runtime/test_004_weight_streaming.py:336]: arg_inputs instead of inputs

File: tests/py/dynamo/runtime/test_004_weight_streaming.py [hunk 2]
```diff
@@ -290,10 +291,6 @@ def test_weight_streaming_cudagraphs(self, _, use_python_runtime):
             ("cpp_runtime", False),
         ]
     )
-    @unittest.skipIf(
```
Comment (keehyuna) [inline tests/py/dynamo/runtime/test_004_weight_streaming.py:293]: this test took 80sec on rtx4080

Comment (narendasan) [reply tests/py/dynamo/runtime/test_004_weight_streaming.py:293]: Feel free to create a new runner instance for tests like these if you feel its taking too long

Comment (keehyuna) [reply tests/py/dynamo/runtime/test_004_weight_streaming.py:293]: I think it looks fine. according to log, it was running under `Test torch compile backend` test, which took 12 minutes.

**Review by peri044 — APPROVED** [review]
LGTM

**Comment by narendasan** [comment]
@keehyuna can you resolve the conflict then we can merge?

**Review by zewenli98 — APPROVED** [review]
LGTM!
